scripts/gen_layer.py

```diff
@@ -0,0 +1,3723 @@
+        gen += '\n\n    return true;\n'
+        gen += '}\n'
+        return gen
+
+    def generate_get_property(self):
+        gen = '\nbool JsonLoader::GetProperty(const Json::Value &props, const std::string &name) {\n'
+        gen += '    const Json::Value &property = props[name];\n\n'
+        gen += self.generate_get_struct('property', 'VkPhysicalDeviceProperties2', self.additional_properties)
+        gen += '\n\n    return true;\n'
+        gen += '}\n'
+        return gen
+
+    def generate_get_struct(self, struct, extends, additional):
+        gen = ''
+        first = True
+        for name, value  in registry.structs.items():
+            if (extends in value.extends and value.isAlias == False) or (name in additional):
+                if self.from_skipped_extension(name, registry):
+                    continue
+                aliases = self.find_aliases(name)
+                aliases.append(name)
+                while (aliases):
+                    current = aliases.pop()
+                    if first:
+                        first = False
+                        gen += '    '
+                    else:
+                        gen += ' else '
+                    gen += 'if (name == \"' + current + '\"'
+                    copy_aliases = aliases
+                    for alias in copy_aliases:
+                        same_version = registry.structs[current].definedByVersion and registry.structs[alias].definedByVersion
+                        same_extension = registry.structs[current].definedByExtensions and registry.structs[current].definedByExtensions == registry.structs[alias].definedByExtensions
+                        if same_version or same_extension:
+                            gen += ' || name == \"' + alias + '\"'
+                            aliases.remove(alias)
+                            
+                    gen += ') {\n'
+
+                    if current in self.non_modifiable_structs:
+                        gen += '        return GetValue' + current[2:] + '(prop);\n'
+                    else:
+                        version = registry.structs[current].definedByVersion
+                        if version:
+                            if version and (version.major != 1 or version.minor != 0):
+                                gen += '        if (!CheckVersionSupport(' + self.get_api_version(registry.structs[current].definedByVersion) + ', name)) return false;\n'
+                        else:
+                            ext = registry.extensions[registry.structs[current].definedByExtensions[0]]
+                            if not ext.name in self.emulated_extensions:
+                                ext_name = ext.upperCaseName + '_EXTENSION_NAME'
+                                gen += '        auto support = CheckExtensionSupport(' + ext_name + ', name);\n'
+                                gen += '        if (support != ExtensionSupport::SUPPORTED) return valid(support);\n'
```

Comment:
This code generates the necessary checks for each structure, if it has the required version or extension, but since the profiles layer can emulated the portability extension, the check is not required here.